streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -369,6 +369,7 @@ public static StreamThread create(final InternalTopologyBuilder builder,
 
         final Consumer<byte[], byte[]> mainConsumer = clientSupplier.getConsumer(consumerConfigs);
         changelogReader.setMainConsumer(mainConsumer);
+        changelogReader.setAdminClient(adminClient);
```

Comment:
Q: Why do you not pass the admin client in the constructor?

Reply:
Yeah that's the plan, I was just hacking it around so that I do not need to change 30+ unit tests. Once it is confirmed to fix the issue I will refactor this PR.